### PR TITLE
feat: Topology Processor

### PR DESCRIPTION
**Pull request by colelaven**
### Proposed Change
Adds topology processor used for gateway topology project
##### Checklist
- [x] Changes are tested
- [x] CI has passed

### CODE REVIEW

File: processor/topologyprocessor/README.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+| `interval`           | duration  | `1m`    | The interval at which topology data is sent to Bindplane via OpAMP.                                                                                                       |
+| `organizationID`     | string    |         | The Organization ID of the Bindplane configuration where this processor is running.                                                                                       |
+| `accountID`          | string    |         | The Account ID of the Bindplane configuration where this processor is running.                                                                                            |
+| `configuration`      | string    |         | The name of the Bindplane configuration this processor is running on.                                                                                                     |
+
+
+### Example configuration
+
+The example configuration below shows ingesting logs and sampling the size of 50% of the OTLP log objects.
```
Comment (dpaasman00) [inline processor/topologyprocessor/README.md:23]: Think this description is a carry over from a different component.

File: processor/topologyprocessor/go.mod [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+	github.com/cespare/xxhash/v2 v2.3.0 // indirect
+	github.com/davecgh/go-spew v1.1.2-0.20180830191138-d8f796af33cc // indirect
+	github.com/go-logr/logr v1.4.2 // indirect
+	github.com/go-logr/stdr v1.2.2 // indirect
+	github.com/gogo/protobuf v1.3.2 // indirect
+	github.com/google/uuid v1.6.0 // indirect
+	github.com/json-iterator/go v1.1.12 // indirect
+	github.com/modern-go/concurrent v0.0.0-20180306012644-bacd9c7ef1dd // indirect
+	github.com/modern-go/reflect2 v1.0.2 // indirect
+	github.com/observiq/bindplane-agent/internal/topology v0.0.0-00010101000000-000000000000
```
Comment (dpaasman00) [inline processor/topologyprocessor/go.mod:28]:
```suggestion
	github.com/observiq/bindplane-agent/internal/topology v1.67.0
```

File: processor/topologyprocessor/go.mod [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+	go.uber.org/multierr v1.11.0 // indirect
+	golang.org/x/net v0.28.0 // indirect
+	golang.org/x/sys v0.26.0 // indirect
+	golang.org/x/text v0.17.0 // indirect
+	google.golang.org/genproto/googleapis/rpc v0.0.0-20240822170219-fc7c04adadcd // indirect
+	google.golang.org/protobuf v1.35.1 // indirect
+	gopkg.in/yaml.v3 v3.0.1 // indirect
+)
+
+replace github.com/observiq/bindplane-agent/internal/report => ../../internal/report
```
Comment (dpaasman00) [inline processor/topologyprocessor/go.mod:54]: Not sure I understand why this replace statement is here? I don't think I saw it anywhere in this go.mod?

**Review by dpaasman00 — CHANGES_REQUESTED** [review]
First pass, some comments, otherwise looking good.

File: extension/bindplaneextension/extension.go [hunk 4]
```diff
@@ -31,43 +33,64 @@ import (
...
+		select {
+		case <-ctx.Done():
+			return nil
+		case b.topologyInterval = <-b.topologyRegistry.SetIntervalChan():
+			if b.topologyInterval > 0 {
+				b.wg.Add(1)
+				go b.reportTopologyLoop()
+			}
+			return nil
+		}
```
[the remark this answers is not in the document]
Comment (colelaven) [reply extension/bindplaneextension/extension.go:82]: thanks for the explanation, that makes sense

File: processor/topologyprocessor/go.mod (hunk 3 above)
Comment (colelaven) [reply processor/topologyprocessor/go.mod:54]: copy paste error 😅

File: processor/samplingprocessor/processor_test.go
Comment (dpaasman00) [inline processor/samplingprocessor/processor_test.go:1]: Curious about this file?

Comment (colelaven) [reply processor/samplingprocessor/processor_test.go:1]: Yeah these tests were being flaky for me, I assume due to the randomness of the sampling processor, so I increased the amount of records in each test and it seems to be way more reliable.
Ik its a bit of scope creep so if you think it'd be better to take out of this PR I will 👍

File: processor/topologyprocessor/README.md [hunk 5]
```diff
@@ -0,0 +1,48 @@
+# Topology Processor
+This processor utilizes request headers to provide extended topology functionality in BindPlane.
+
+## Minimum agent versions
+- Introduced: [v1.67.0](https://github.com/observIQ/bindplane-otel-collector/releases/tag/v1.6.7)
```
Comment (dpaasman00) [inline processor/topologyprocessor/README.md:5]:
```suggestion
- Introduced: [v1.68.0](https://github.com/observIQ/bindplane-otel-collector/releases/tag/v1.68.0)
```

File: go.mod [hunk 6]
```diff
@@ -199,6 +199,8 @@ require (
 )
 
 require (
+	github.com/observiq/bindplane-otel-collector/internal/topology v1.67.0
+	github.com/observiq/bindplane-otel-collector/processor/topologyprocessor v1.67.0
```
Comment (dpaasman00) [inline go.mod:203]:
```suggestion
	github.com/observiq/bindplane-otel-collector/internal/topology v1.68.0
	github.com/observiq/bindplane-otel-collector/processor/topologyprocessor v1.68.0
```

File: processor/samplingprocessor/processor_test.go
Comment (dpaasman00) [reply processor/samplingprocessor/processor_test.go:1]: No I appreciate it, thanks for doing that!

**Pull request merged**